hazelcast-sql/src/main/java/com/hazelcast/sql/impl/expression/service/GetDdlFunction.java

```diff
@@ -87,10 +79,14 @@ public String eval(Row row, ExpressionEvalContext context) {
         if (obj == null) {
             throw QueryException.error("Object '" + objectName + "' does not exist in namespace '" + namespace + "'");
         } else if (obj instanceof SqlCatalogObject) {
-            if (obj instanceof Mapping) {
-                NodeEngine nodeEngine = context.getNodeEngine();
-                if (nodeEngine.getConfig().getSecurityConfig().isEnabled()) {
-                    throw new AccessControlException("GET_DDL is disabled for mapping in secure environment");
+            SqlCatalogObject catalogObject = (SqlCatalogObject) obj;
+            // TODO: view mapping/view 'view'/view type?
+            if (catalogObject instanceof DataConnectionCatalogEntry) {
+                context.checkPermission(new SqlPermission(catalogObject.name(), ACTION_VIEW_DATACONNECTION));
+            } else {
+                if (context.isSecurityEnabled()) {
+                    throw new UnsupportedOperationException("GET_DDL is not available "
```

Comment:
for mapping we should check permissions and views and types can be visible, because they do not contain any secrets. Limiting visibility of SQL objects in general is outside of the scope of this issue.

Reply:
I don't see change enabling GET_DDL for mappings

Reply:
hmm, we do not have permission to view mappings

Reply:
https://hazelcast.atlassian.net/browse/HZ-3190